Arm/Memory/MemoryProofs.lean

```diff
@@ -886,65 +862,45 @@ theorem entire_memory_subset_legal_regions_eq_addr
   simp_all [mem_subset, mem_legal]
   bv_omega
 
-private theorem read_mem_bytes_of_write_mem_bytes_subset_n2_eq_alt_helper (val : BitVec (x * 8))
-  (h0 : 0 < x)
-  (h : (BitVec.toNat (addr2 - addr2) + x) * 8 - 1 -
-         BitVec.toNat (addr2 - addr2) * 8 + 1
-        =
-       x * 8) :
+private theorem read_mem_bytes_of_write_mem_bytes_subset_n2_eq_alt_helper (val : BitVec (x * 8)):
   val =
-    BitVec.cast h
-      (extractLsb ((BitVec.toNat (addr2 - addr2) + x) * 8 - 1)
-        (BitVec.toNat (addr2 - addr2) * 8) val) := by
+    extractLsb' ((BitVec.toNat (addr2 - addr2)) * 8) (x * 8) val := by
   ext
```

Comment:
This theorem feels like it shouldn't be needed anymore:
- `addr2 - addr2` should simplify to `0`, via `BitVec.sub_self`
- `(BitVec.toNat 0) * 8` should constant fold to just `0`
- The final bit I'm not sure we have a simp-lemma for already, but `extractLsb' 0 w val`, where `val : BitVec w` should simplify to just `val`

Thus, if we add the last, simpler, simp-lemma, that should subsume this one

Reply:
You are right! The new proof is:
```
simp only [BitVec.sub_self, toNat_ofNat, Nat.reducePow, Nat.zero_mod, Nat.zero_mul]
exact Eq.symm (extractLsb'_eq val)
```